AncestralAllele plugin

willmclaren opened the pull request:
retrieves ancestral alleles for input variants from FASTA file

helensch commented on AncestralAllele.pm:
```diff
@@ -0,0 +1,277 @@
+  return $self->{_index_type};
+}
+
+## CRIBBED FROM Bio::EnsEMBL::Variation::Utils::FastaSequence
```
I suggest this  the code based on existing module is moved to a shared utils space.

helensch commented on AncestralAllele.pm:
```diff
@@ -0,0 +1,277 @@
+  return $self->{_name_map};
+}
+
+1;
```
Minor comment. Suggest putting a line feed here.

helensch commented on AncestralAllele.pm:
```diff
@@ -0,0 +1,277 @@
+      $sr_name =~ /^chr/i ? (substr($sr_name, 3)) : ('chr'.$sr_name, 'CHR'.$sr_name)
+    ) {
+      if($map->{$alt}) {
+        print STDERR "USING SYNOYM $alt FOR $sr_name\n" if $DEBUG;
```
Typo in SYNONYM

helensch commented:
It might be useful to document when "?" and "-" are returned as ancestral alleles.

willmclaren commented:
Thanks, addressed your comments.

The code won't be moved to a shared utils module yet as this will require modifications to the ensembl-variation API.

willmclaren commented:
@helensch I think I addressed all of your comments now, are you happy to merge or shall I nominate another?
